client-core/src/key/public_key.rs

```diff
@@ -74,7 +74,7 @@ impl From<PublicKey> for SecpPublicKey {
 
 impl From<&PublicKey> for SecpPublicKey {
     fn from(public_key: &PublicKey) -> SecpPublicKey {
-        public_key.0
+        public_key.0.clone()
```

Comment:
is clone necessary?

Reply:
Yes, with "zeroize" feature, the macro for internal library structs won't derive `Copy` (as some of them implemented `Drop`)... for public key, it won't be necessary, but it'd require a bit more complex changes